src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -66,23 +76,81 @@ public async Task<BundleResponse> Handle(BundleRequest bundleRequest, Cancellati
         {
             var bundleResource = bundleRequest.Bundle.ToPoco<Hl7.Fhir.Model.Bundle>();
 
-            if (bundleResource.Type != Hl7.Fhir.Model.Bundle.BundleType.Batch)
+            await FillRequestLists(bundleResource.Entry);
+
+            if (bundleResource.Type == Hl7.Fhir.Model.Bundle.BundleType.Batch)
+            {
+                var responseBundle = new Hl7.Fhir.Model.Bundle
+                {
+                    Type = Hl7.Fhir.Model.Bundle.BundleType.BatchResponse,
+                };
+
+                await ExecuteAllRequests(responseBundle);
+                return new BundleResponse(responseBundle.ToResourceElement());
+            }
+            else if (bundleResource.Type == Hl7.Fhir.Model.Bundle.BundleType.Transaction)
             {
-                throw new MethodNotAllowedException(Microsoft.Health.Fhir.Api.Resources.OnlyCertainBundleTypesSupported);
```

Comment:
If we're no longer using `OnlyCertainBundleTypesSupported` can you remove it?

Reply:
It has already been removed.